spec/zoneddatetime.html

```diff
@@ -1456,28 +1450,31 @@ <h1>
           1. Repeat, while _days_ &gt; 0 and ℝ(_relativeResult_.[[EpochNanoseconds]]) &gt; _endNs_,
             1. Set _days_ to _days_ - 1.
             1. Set _relativeResult_ to ? AddDaysToZonedDateTime(_startInstant_, _startDateTime_, _timeZoneRec_, _zonedRelativeTo_.[[Calendar]], _days_).
-        1. Set _nanoseconds_ to _endNs_ - ℝ(_relativeResult_.[[EpochNanoseconds]]).
+        1. Set _norm_ to NormalizedTimeDurationFromEpochNanosecondsDifference(_endNs_, _relativeResult_.[[EpochNanoseconds]]).
         1. Let _done_ be *false*.
         1. Let _dayLengthNs_ be ~unset~.
         1. Repeat, while _done_ is *false*,
           1. Let _oneDayFarther_ be ? AddDaysToZonedDateTime(_relativeResult_.[[Instant]], _relativeResult_.[[DateTime]], _timeZoneRec_, _zonedRelativeTo_.[[Calendar]], _sign_).
-          1. Set _dayLengthNs_ to ℝ(_oneDayFarther_.[[EpochNanoseconds]] - _relativeResult_.[[EpochNanoseconds]]).
-          1. If (_nanoseconds_ - _dayLengthNs_) &times; _sign_ &ge; 0, then
-            1. Set _nanoseconds_ to _nanoseconds_ - _dayLengthNs_.
+          1. Set _dayLengthNs_ to NormalizedTimeDurationFromEpochNanosecondsDifference(_oneDayFarther_.[[EpochNanoseconds]], _relativeResult_.[[EpochNanoseconds]]).
+          1. Let _oneDayLess_ be ? SubtractNormalizedTimeDuration(_norm_, _dayLengthNs_).
```

Comment:
Is this call really fallible? `dayLengthNs` can't exceed the maximum instant difference (2 × 8.64 × 10<sup>12</sup> seconds), so it's always a valid normalised time duration. The difference `norm - dayLengthNs` also can't exceed the maximum instant difference, see <https://searchfox.org/mozilla-central/rev/68b1b0041a78abd06f19202558ccc4922e5ba759/js/src/builtin/temporal/ZonedDateTime.cpp#840-876>. Or am I missing something?

Reply:
I don't think you're missing anything.